src/utils.jl

```diff
@@ -73,3 +73,23 @@ function testfeaturecollection(fc)
     @assert geometrycolumns(fc) isa NTuple "feature collection $fc doesn't return a `NTuple` from `geometrycolumns`."
     return true
 end
+
+"""
+    testraster(raster)
+
+Test whether the required interface for your `raster` has been implemented correctly.
+"""
+function testraster(raster)
+    @assert israster(typeof(raster)) "$raster doesn't implement `israster`."
+    @assert trait(raster) isa AbstractRasterTrait "$raster does not return an `AbstractRasterTrait` for `trait`."
+    @assert isnothing(geomtrait(raster))
+    @assert !isnothing(crs(raster)) "Raster $raster doesn't return anything for `crs`"
```

Comment:
CRS is a little complicated for netcdf - a raster can be projected in one CRS but the bounding box and index can be in another, usually EPSG::4326. In Julia often we don't know the underlying crs either, as there is no parser for the CF crs format.

In Rasters.jl there are two crs fields for this reason.

Reply:
I will document this further, but it should be the crs for the coordinates returned from affine. Your netcdf essentially holds two rasters definitions.

Reply:
Yeah exactly, they have to match, and there are two definitions.